lib/ecto/migration/runner.ex

```diff
@@ -214,9 +213,14 @@ defmodule Ecto.Migration.Runner do
     end)
   end
 
+  defp log_sql do
+    Agent.get(runner(), fn %{log_sql: log_sql} -> log_sql end)
+  end
+
   defp log_and_execute_ddl(repo, level, command) do
     log(level, command(command))
-    repo.__adapter__.execute_ddl(repo, command, @opts)
+    opts = [timeout: :infinity, log: log_sql]
```

Comment:
Let's read the `log_sql` value in `repo_and_direction_and_level/0` function and rename it to `runner_config()` and pass the configuration as arguments to this function.